pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/writer/impl/FixedBitMVForwardIndexWriter.java

```diff
@@ -76,7 +76,7 @@ public class FixedBitMVForwardIndexWriter implements Closeable {
 
   public FixedBitMVForwardIndexWriter(File file, int numDocs, int totalNumValues, int numBitsPerValue)
       throws IOException {
-    float averageValuesPerDoc = totalNumValues / numDocs;
+    double averageValuesPerDoc = Math.max(totalNumValues / (double) numDocs, 1f);
```

Comment:
Fixing the integer division and also using max to ensure averageValuesPerDoc is never 0!